packages/database/src/migrations/20201117053359-AddPsssWoWIncreaseIndicators-modifies-data.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+import { arrayToDbString, generateId, insertObject } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const CONDITION_CODES = ['AFR', 'DIA', 'ILI', 'PF', 'DLI'];
+
+// eg 'PSSS_Confirmed_AFR_Site_Average'
+const psssCode = (descriptor, condition = '', isConfirmed = false) => {
+  const parts = ['PSSS', descriptor];
+  if (condition) {
+    parts.splice(1, 0, condition);
+  }
+  if (isConfirmed) {
+    parts.splice(1, 0, 'Confirmed');
+  }
+
+  return parts.join('_');
+};
+
+const insertIndicator = async (db, indicator) => {
+  await insertObject(db, 'indicator', { ...indicator, id: generateId() });
+  await insertObject(db, 'data_source', {
+    id: generateId(),
+    code: indicator.code,
+    type: 'dataElement',
+    service_type: 'indicator',
+  });
+};
```

Comment:
This seems really handy, as I'm sure many of us may forget to create the `data_source` for the indicator when creating one. Think it's worth moving this to migration utils?

Reply:
That's a great point @rohan-bes ! And as a matter of fact I initially forgot to add the data sources, even though I've done this in the past. Then I lost time trying to understand why my data is not fetched properly 😆


The problem with migrations is that they are historically bound to the state of the schema when they run. This means that the same util may not work in the future (eg if we break `data_source` in two tables: `data_element` and `data_group`)

I'm still happy to add this to the `Utility Templates` section of the migration utils, as a source of documentation.

Idea: we could establish a new pattern: "versioned" functions, similar to versioned apis. Those could be either
`insertIndicatorV1`, or `insetIndicators(args, version)` where different logic is executed according to the specified version. I don't know if it's worth the maintenance overhead, just bringing this up for brainstorming.